src/main/webapp/scripts/session-script.js

```diff
@@ -85,6 +122,82 @@ function setReadOnlyInputs(sessionId) {
   welcomeMessageInput.readOnly = true;
 }
 
+/*
+ * function updateUI() refreshes information client side, 
+ * updating the UI in checking for new attendees and for
+ * whoever the controller is.
+ */
+function updateUI() {
+  setInterval(() => {
+    serverClient.getSession().then(session => {
+      const /** Array */ listOfAttendeeScreenNames = [];
+      session.getListOfAttendees().forEach(attendee => {
+        listOfAttendeeScreenNames.push(attendee.getScreenName());
+      });
+      updateSessionAttendees(listOfAttendeeScreenNames,
+          session.getScreenNameOfController());
+      updateController(session.getScreenNameOfController());
+    }).catch(error => {
+      window.alert('No contact with the server!');
+    });
+  }, SESSION_REFRESH_CADENCE_MS);
+}
+
+/**
+ * function updateSessionAttendees() adds new attendees to the
+ * session to the session info attendee div. Also removes attendees 
+ * if they left the session. Alerts users of anyone who has left/entered.
+ * @param {Array} updatedAttendees array of new attendees
+ * @param {string} controller name of the controller of the session
+ */
+function updateSessionAttendees(updatedAttendees, controller) {
+  const /** Array */ newAttendees = updatedAttendees.filter(attendee => {
+    return !currentAttendees.includes(attendee);
+  });
+  const /** Array */ attendeesThatHaveLeft = 
+      currentAttendees.filter(attendee => {
+        return !updatedAttendees.includes(attendee);
+  });
+  if (newAttendees.length > 0) {
+    let /** string */ displayMessage =
+        'The following people have joined the session: ';
+    displayMessage += newAttendees.join(', ');
+    if (attendeesThatHaveLeft.length > 0) {
+      displayMessage += '. The following people have left the session: ';
+      displayMessage += attendeesThatHaveLeft.join(', ');
+    }
+    notifyOfChangesToMembership(displayMessage);
+  } else if (newAttendees.length === 0 && attendeesThatHaveLeft.length 
```

Comment:
I *think* you can get rid of this case if you just add a the period and a newline character after line 164 and remove the space and period from line 167

Reply:
yup it works